File: Patterns/Command Pattern/CSharp/Command/Light/Light.cs
using System;

namespace Command.Light
{
    public class Light
    {
        private string location;
        public int State { get; private set; }

        public const int ON = 1;
        public const int OFF = 0;

        public Light(string location)
        {
            State = OFF;
            this.location = location;
        }

        public void On()
        {
            State = ON;
            Console.WriteLine($"{location} 조명이 켜졌습니다.");
        }

        public void Off()
        {
            State = OFF;
            Console.WriteLine($"{location} 조명이 꺼졌습니다.");
        }
    }
}

File: Patterns/Command Pattern/Cpp/Command/Light/Light.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class Light {
private:
    string location;
    int state;

public:
    const int ON = 1;
    const int OFF = 0;

    Light(string location) : location{location}, state{OFF} {}

    void on() {
        state = ON;
        cout << location << " 조명이 켜졌습니다.\n"; 
    }

    void off() { 
        state = OFF;
        cout << location << " 조명이 꺼졌습니다.\n"; 
    }

    int getState() { return state; }
};